pkg/operation/botanist/component/kubeapiserverexposure/kube_apiserver_service.go

```diff
@@ -146,6 +151,10 @@ func (s *service) Deploy(ctx context.Context) error {
 		if s.values.gardenerManaged {
 			metav1.SetMetaDataLabel(&obj.ObjectMeta, v1beta1constants.LabelAPIServerExposure, v1beta1constants.LabelAPIServerExposureGardenerManaged)
 		}
+		if s.values.topologyAwareRoutingEnabled {
+			metav1.SetMetaDataAnnotation(&obj.ObjectMeta, corev1.AnnotationTopologyAwareHints, "auto")
+			metav1.SetMetaDataLabel(&obj.ObjectMeta, resourcesv1alpha1.EndpointSliceHintsConsider, "true")
+		}
```

Comment:
I think for this component the annotation and label is not reverted if the feature gate is turned off again because both are merged a few lines above. In some components, we calculate them from the beginning but in others like this, we retain the already existing ones.

Reply:
Good point. It should be now addressed.